src/gui/dialogs/units_dialog.hpp

```diff
@@ -0,0 +1,232 @@
+	std::string variation() const
+	{
+		return variation_;
+	}
+
+	units_dialog& set_title(const std::string& title) {
+		title_ = title;
+		return *this;
+	}
+
+	units_dialog& set_ok_label(const std::string& ok_label) {
+		ok_label_ = ok_label;
+		return *this;
+	}
+
+	units_dialog& show_variations(const bool show_variation) {
+		show_variation_grid_ = show_variation;
+		return *this;
+	}
+
+	units_dialog& show_gender(const bool show_gender) {
+		show_gender_grid_ = show_gender;
+		return *this;
+	}
+
+	units_dialog& show_dismiss_option(const bool show_dismiss) {
+		show_dismiss_ = show_dismiss;
+		return *this;
+	}
+
+	units_dialog& show_rename_option(const bool show_rename) {
+		show_rename_ = show_rename;
+		return *this;
+	}
+
+	units_dialog& show_all_headers() {
+		visible_headers_.set();
+		return *this;
+	}
+
+	units_dialog& hide_all_headers() {
+		visible_headers_.reset();
+		return *this;
+	}
+
+	units_dialog& show_header(const int col_num, const bool visible = true) {
+		visible_headers_[col_num] = visible;
+		return *this;
+	}
+
+	units_dialog& set_help_topic(const std::string& topic_id) {
+		topic_id_ = topic_id;
+		return *this;
+	}
+
+	units_dialog& set_team(team* team) {
+		team_ = team;
+		return *this;
+	}
+
+	units_dialog& set_types(std::vector<const unit_type*> types) {
+		recruit_list_ = types;
+		return *this;
+	}
+
+	units_dialog& set_units(std::vector<unit_const_ptr> units) {
+		recall_list_ = units;
+		return *this;
+	}
+
+	/**
+	 * Corresponding to each widget in the row with id 'id', there is a lambda that generates
+	 * the corresponding label to set to that widget. This method sets those generator functions.
+	 * The 'id's must be the same as those defined for the various widgets
+	 * in the [list_definition] in the WML file.
+	 */
+	units_dialog& set_column_generator(std::string id, const std::function<std::string(const int)> generator);
+
+	/**
+	 * Sets the generator function for the tooltips
+	 */
+	units_dialog& set_tooltip_generator(const std::function<std::string(const int)> generator)
+	{
+		tooltip_gen_ = generator;
+		return *this;
+	}
+
+	template<typename Func>
+	units_dialog& set_sorter(int col, const Func& sorter)
+	{
+		find_widget<gui2::listbox>("recall_list").register_sorting_option(col, sorter);
+		return *this;
+	}
+
+	units_dialog& set_translatable_sorter(int col, const std::function<std::string(const int)> sorter)
+	{
+		find_widget<gui2::listbox>("recall_list").register_translatable_sorting_option(col, sorter);
+		return *this;
+	}
+
+private:
+	std::vector<const unit_type*> recruit_list_;
+	std::vector<unit_const_ptr> recall_list_;
+
+	team* team_;
+
+	int selected_index_;
+
+	std::string title_;
+	std::string ok_label_;
+	std::string topic_id_;
+	bool show_variation_grid_;
+	bool show_gender_grid_;
+	bool show_dismiss_;
+	bool show_rename_;
+
+	bool show_image_;
+
+	boost::dynamic_bitset<> visible_headers_;
+	std::map< std::string, std::function<std::string(const int)> > column_generators_;
```

Comment:
Formatting - excess spaces.

Reply:
removed extra space after < and before >.